ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -1876,24 +1871,17 @@ OGRErr OGRFlatGeobufLayer::ICreateFeature(OGRFeature *poNewFeature)
         if (c == 0)
             return CPLErrorIO("writing feature");
         if (m_bCreateSpatialIndexAtClose) {
-            const auto item = std::make_shared<FeatureItem>();
-#if defined(__MINGW32__) && __GNUC__ >= 7
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wnull-dereference"
-#endif
-            item->size = static_cast<uint32_t>(fbb.GetSize());
-            item->offset = m_writeOffset;
-            item->nodeItem = {
+            FeatureItem item;
+            item.size = static_cast<uint32_t>(fbb.GetSize());
+            item.offset = m_writeOffset;
+            item.nodeItem = {
                 psEnvelope.MinX,
                 psEnvelope.MinY,
                 psEnvelope.MaxX,
                 psEnvelope.MaxY,
                 0
             };
-#if defined(__MINGW32__) && __GNUC__ >= 7
-#pragma GCC diagnostic pop
-#endif
-            m_featureItems.push_back(item);
+            m_featureItems.emplace_back(std::move(item));
```

Comment:
Instead of moving an initialized item into `emplace_back`, is there now way to avoid the initial item on the stack?

Reply:
Browsing through https://stackoverflow.com/questions/1902832/resize-versus-push-back-in-stdvector-does-it-avoid-an-unnecessary-copy-assig makes me think that the following alternative would be more or less equivalent to the current state of the PR. Actually trying it I can't see any measurable timing difference:
```
diff --git a/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp b/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp
index 5d3fb3cd18..df3ae9a8e3 100644
--- a/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp
+++ b/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp
@@ -1871,7 +1871,8 @@ OGRErr OGRFlatGeobufLayer::ICreateFeature(OGRFeature *poNewFeature)
         if (c == 0)
             return CPLErrorIO("writing feature");
         if (m_bCreateSpatialIndexAtClose) {
-            FeatureItem item;
+            m_featureItems.resize(m_featureItems.size() + 1);
+            FeatureItem& item = m_featureItems.back();
             item.size = static_cast<uint32_t>(fbb.GetSize());
             item.offset = m_writeOffset;
             item.nodeItem = {
@@ -1881,7 +1882,6 @@ OGRErr OGRFlatGeobufLayer::ICreateFeature(OGRFeature *poNewFeature)
                 psEnvelope.MaxY,
                 0
             };
-            m_featureItems.emplace_back(std::move(item));
         }
         m_writeOffset += c;
 
```